pkg/cmd/admin/policy/modify_roles.go

```diff
@@ -371,9 +390,16 @@ subjectCheck:
 	}
 
 	if isUpdate {
+		if len(o.RoleBindingName) > 0 {
+			roleBinding.Name = o.RoleBindingName
+		}
 		err = o.RoleBindingAccessor.UpdateRoleBinding(roleBinding)
 	} else {
-		roleBinding.Name = getUniqueName(o.RoleName, roleBindingNames)
+		if len(o.RoleBindingName) > 0 {
```

Comment:
I'd like to see all this logic hoisted up to the top

```
if name provided {
  look up named binding
  if exists and points to other roleref, error
  if exists, set isUpdate=true
  if doesn't exist, populate object with specified name, set isUpdate=false
} else {
  get bindings to role
  if exists, pick first to modify, set isUpdate=true
  if doesn't exist, get existing names and generate unique name, set isUpdate=false
}
```

everything from that point on can deal with the rolebinding the same way in either case